### PR TITLE
More clean up of LibuvConnection

### PR DESCRIPTION
**Pull request by davidfowl**
- Use C# 7
- Use Buffer<T>.Pin to get access to the underlying
pointer instead of using TryGetPointer.

### CODE REVIEW

File: src/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv/Internal/LibuvConnection.cs [hunk 1]
```diff
@@ -107,16 +108,12 @@ private static LibuvFunctions.uv_buf_t AllocCallback(UvStreamHandle handle, int
 
         private unsafe LibuvFunctions.uv_buf_t OnAlloc(UvStreamHandle handle, int suggestedSize)
         {
-            Debug.Assert(_currentWritableBuffer == null);
```
Comment (cesarbs) [inline src/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv/Internal/LibuvConnection.cs]: Why did you remove the `Assert`?

Comment (davidfowl) [inline src/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv/Internal/LibuvConnection.cs]: Didn't seem necessary. @pakrym originally added it when we ported.

Comment (davidfowl) [inline src/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv/Internal/LibuvConnection.cs]: Added it back. It can't hurt.
